include/libtorrent/session_handle.hpp

```diff
@@ -449,6 +449,14 @@ namespace libtorrent
 		// posted, regardless of the alert mask.
 		void dht_live_nodes(sha1_hash const& nid);
 
+		// Query the DHT to retrieve a sample of the infohashes that the specific
+		// node currently have in their storage.
+		// The ``target`` is included for iterative lookups so that indexing nodes
+		// can perform a keyspace traversal with a single RPC per node by adjusting
+		// the target value for each RPC. It has no effect on the returned sample value.
+		// The result is posted as a ``dht_sample_infohashes_alert``.
+		void dht_sample_infohashes(udp::endpoint const& ep, sha1_hash const& target);
```

Comment:
sorry for the delay. what did we conclude here? Without having been too involved in how this traversal should work, my guess (from looking at this API and documentation) would be that I am (as a client) supposed to call this function repeatedly, adjusting the ``target`` argument to be evenly distributed across the key space.

The conflict with that notion is that you say "It has no effect on the returned sample value.". If that is indeed true, why do I need to pass it in here? Why wouldn't libtorrent randomize it (or whatever else it could do to determine it)?

Furthermore, thinking about how I imagine this to be implemented is that instead of narrowing in on a certain info-hash when traversing the DHT, I would expect every lookup to adjust the target info-hash some way, to try to spread out the nodes being visited. In either case, I don't understand why I'm passing in ``target`` here.